### PR TITLE
Add peer timeout test

### PR DESCRIPTION
**Pull request by tomgroenwoldt**
This PR adds an integration test for the peer timeout functionality.

Please merge #24 before this.

### CODE REVIEW

**Comment by tomgroenwoldt** [comment]
Adding another peer resolved the weird behavior.

**Review by svenstaro — CHANGES_REQUESTED** [review]
Could you make a separate PR for the commit that changes the network setup?

File: Cargo.toml [hunk 1]
```diff
@@ -34,6 +34,7 @@ tracing = { version = "0.1", features = ["log"] }
 tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 wireguard-keys = "0.1"
 humantime = "2.1.0"
+rand = "0.8.5"
```
[the remark this answers is not in the document]
Comment (tomgroenwoldt) [reply Cargo.toml]: Done.

**Review by svenstaro — APPROVED** [review]

File: tests/integration.rs [hunk 2]
```diff
@@ -327,3 +328,125 @@ async fn join_network(
...
+    assert_eq!(consul_peers.len(), 3);
+
+    // Kill a random peer.
+    let mut rng = rand::thread_rng();
+    peers.shuffle(&mut rng);
+    let (random_peer, remaining_peers) = peers.split_first().expect("Peers are empty.");
+
+    // This sleep statement is just for demonstration purposes:
+    // The peer timeout is triggered for everyone via this statement.
+    sleep(Duration::from_secs(4)).await;
```
[the remark this answers is not in the document]
Comment (tomgroenwoldt) [reply tests/integration.rs:429]: Oh yeah, this was just for keeping the test running and originally had 2000s. I forgot do delete it.

File: tests/integration.rs [hunk 3]
```diff
@@ -327,3 +328,125 @@ async fn join_network(
...
+
+    sleep(Duration::from_secs(1)).await;
+
+    let consul_peers = consul.client.get_peers().await?;
+    assert_eq!(consul_peers.len(), 3);
+
+    // Kill a random peer.
+    let mut rng = rand::thread_rng();
+    peers.shuffle(&mut rng);
+    let (random_peer, remaining_peers) = peers.split_first().expect("Peers are empty.");
```
[the remark this answers is not in the document]
Comment (tomgroenwoldt) [reply tests/integration.rs:425]: Sounds reasonable.

**Review by svenstaro — APPROVED** [review]

**Comment by svenstaro** [comment]
Great job!

**Pull request merged**